### DISCUSSION TITLE
OSM Planet Data Required Disk Space

### DISCUSSION THREAD

**Discussion by mertbakir** [valhalla/valhalla#3661] [Q&A] [answered]
Hi, I wonder how much disk space required for valhalla to process whole planet data? Do we know like roughly speaking 1 gb osm.pbf will become 3 gb when it's processed?

**Comment by nilsnolde** [comment, marked as answer]
strongly depends what build options you use exactly, and also where the data is coming from (seems like osm in your case). 

for osm roughly:
- graph without historical traffic: ~ 1 - 1.5 x OSM PBF size (currently around 75 GB planet graph)
- elevation (uncompressed for graph, no oceans except for ferry lines): ~350 GB

**Reply by mertbakir** [reply]
Thanks @nilsnolde , 

Can you name some example data providers other than osm? Free or commercial, so that I can do some research.
I guess geofabrik data by default has no historical traffic. Where can I get osm graph with historical traffic?

**Reply by nilsnolde** [reply]
tomtom (my personal preference) or here usually. if you want you can [let me know](enquiry@gis-ops.com) once you want to explore adding commercial data, we provide quite a few commercial services around that topic (converting "any" data to OSM, adding historical traffic etc).

there's no osm with historical traffic AFAIK (maybe mapbox has smth by now? kinda doubt it though, probably only live traffic if any)

**Reply by mertbakir** [reply]
One more question, whats the avg. expected duration, for valhalla to build all the tiles and start running ready if it parses planet osm data ~75 GB?

**Reply by kevinkreiser** [reply]
depends on the number of cpus you let it use. for a 32 core machine it takes about a day but almost a 3rd of that is spent in the parsing phase. the next heaviest phase is the build phase but that can be distributed to multiple machines at once and so you can speed it up quite a bit if you want to run a bunch of machines

**Reply by mertbakir** [reply]
@kevinkreiser Thanks, Is there a documentation for setting up multiple machines? What's the minumum and recommended specs (cpu/ram) for a single machine?

**Reply by kevinkreiser** [reply]
its not documented no. but basically there is a step before `build` called `construct_edges` (maybe a different spelling?) where we create this json file that has a list of all the tiles to build. from there you can simply divide and conquer the `build` step by takign the json splitting it up into multiple jsons (1 per computer you want to run it on) and then running each on each computer for just the build stage (of course each also has to have access to all the other intermedia data that was created in the previous stages so yu have to copy the data around as well. anyway when all the computers are done with their portion of that json manifest you can take all of their outputs put them all back together into one directory structure and continue the rest of the build process on a single machine (the rest of the stages dont take that long maybe 4 hours or something on a 32 core machine)

**Reply by kevinkreiser** [reply]
minimum specs for a single machine are quite low. you can probably run it on one core and 8 gigs of ram but you'd wait an eternity for the results. i routinely run it on a 24 core 32 gig machine with 150gb of nvme and it takes about a day to complete. waiting longer than a day, imho, is too much

**Reply by mertbakir** [reply]
Many Thanks,
